.github/workflows/release.yml

```diff
@@ -24,24 +27,27 @@ jobs:
       - uses: actions/checkout@v2
 
       # Decrypt the ssh secret
-      - name: Decrypt files
+      - name: Decrypt files and set constants
         env:
           SSH_KEY: ${{ secrets.SSH_KEY }}
           MOZILLA_API_SECRET: ${{ secrets.MOZILLA_API_SECRET }}
           MOZILLA_API_KEY: ${{ secrets.MOZILLA_API_KEY }}
-          GHWS: ${{ github.workspace }}
-          # We use version 0.0.0 for testing this script, becuase github.event.release.tag_name
-          # only exists if the release event has fired
-          TAG_NAME: ${{ github.event.release.tag_name } || {'0.0.0'}}
+          TAG_NAME: ${{ github.event.release.tag_name }}
+        shell: bash
         run: |
-          echo "$SSH_KEY" | base64 -d > ./$GHWS/scripts/ssh_key
-          chmod 600 ./$GHWS/scripts/ssh_key
+          echo "$SSH_KEY" > ./scripts/ssh_key
+          chmod 600 ./scripts/ssh_key
+          ls -al ./scripts/
 
       # Set up the environment
       - name: Setup environment
-        run: sudo apt-get --yes install ./$GHWS/ubuntu_touch/*.deb
+        shell: bash
+        run: |
+          sudo apt-get update
+          sudo apt-get --yes install ./scripts/ubuntu_touch_packages/*.deb
 
       # Runs the build scripts
       - name: Run the build scripts
-        # Switch -t indicates a tag release; -d indicates dry run (for testing)
-        run: sh DISPLAY=:99.0 ./$GHWS/scripts/create_all_packages.sh -t -v $TAG_NAME -d
+        shell: bash
+        # Switch -t indicates a tag release; add -d for dry run (for testing)
+        run: DISPLAY=:99.0 ./scripts/create_all_packages.sh -t -v "$TAG_NAME"
```

Comment:
The parameters must be different when it's a nightly build : see .travis.yml

Reply:
Interestingly, if `$TAG_NAME` is empty, then the script builds a nightly build and uploads it to nightly with the correct naming convention (see screenshot below from my test earlier today, taken from http://download.kiwix.org/nightly/2020-12-11/ ). In the light of this, do you still think I should add logic to remove the parameters if it is a nightly build?

![image](https://user-images.githubusercontent.com/4304337/101933162-ba059b80-3bd3-11eb-9781-802c9985a9e0.png)

Reply:
The parameters need to be removed on a nightly build, at least for the firefox extension : depending on the presence of `-t` parameter, it will create a "unsigned listed" or "signed unlisted" extension. They have different ids, and different purposes : 

- "unlisted" extensions are for testing purpose, automatically signed, and not available on Mozilla store
- "listed" extensions are for deployment purpose. They are manually uploaded, checked by Mozilla, and signed by Mozilla if accepted, and afterwards available on their store

Reply:
Ah, OK. I think I can simply do:
```
if [ "x$TAG_NAME" = "x" ]; then
   DISPLAY=:99.0 ./scripts/create_all_packages.sh -t -v "$TAG_NAME"
else
   DISPLAY=:99.0 ./scripts/create_all_packages.sh
fi
```
Because the script will only ever be run with a TAG_NAME (in which case it's a release), or without one (in which case it's a nightly build or else an Action dispatch build). I'll try this but won't be able to test properly till merged to master.

Reply:
Oops, I think the if and else blocks need to be reversed.

Reply:
Done (with the then-else blocks the reverse of the sample above).